back/engines/commercial/report_builder/app/services/report_builder/queries/registrations.rb

```diff
@@ -0,0 +1,67 @@
+module ReportBuilder
+  class Queries::Registrations < ReportBuilder::Queries::Base
+    def run_query(
+      start_at: nil,
+      end_at: nil,
+      resolution: 'month',
+      compare_start_at: nil,
+      compare_end_at: nil,
+      **_other_props
+    )
+      start_date, end_date = TimeBoundariesParser.new(start_at, end_at).parse
+      registrations_in_period = User.where(registration_completed_at: start_date..end_date)
+
+      # Time series
+      registrations_timeseries = registrations_in_period
+        .select("
+          count(*) as registrations,
+          date_trunc('#{resolution}', registration_completed_at) as date_group
+        ")
+        .group('date_group')
+        .order('date_group')
+        .map do |row|
+          {
+            registrations: row.registrations,
+            date_group: row.date_group.to_date
+          }
+        end
+
+      registrations_whole_period = registrations_in_period.count
+
+      response = {
+        registrations_timeseries: registrations_timeseries,
+        registrations_whole_period: registrations_whole_period,
+        registration_rate_whole_period: registration_rate(
+          registrations_whole_period,
+          start_date,
+          end_date
+        )
+      }
+
+      if compare_start_at && compare_end_at
+        registrations_compared_period = User
+          .where(registration_completed_at: compare_start_at..compare_end_at)
+          .count
+
+        response[:registrations_compared_period] = registrations_compared_period
+        response[:registration_rate_compared_period] = registration_rate(
+          registrations_compared_period,
+          compare_start_at,
+          compare_end_at
+        )
+      end
```

Comment:
Why don't we use `TimeBoundariesParser` for these dates, as we do for start_at & end_at on Line 11:
```
start_date, end_date = TimeBoundariesParser.new(start_at, end_at).parse
```
Is this intentional?

Reply:
The reason why I am using for `start_at` and `end_at` is because they might be missing. If they are missing, we still want to continue with our calculations. In this case with the `compare` dates, if they are missing it means that we don't need to calculate anything.